phpfpm module crashes when fastcgi endpoint is down

## Bug report
Using fcgi urls in phpfpm causes telegraf to crash when the endpoint doesnt' exist

### Relevant telegraf.conf:
```
[phpfpm]
  urls = ["fcgi://127.0.0.1:48840/fpmstatus"]
```

### System info:
Telegraf 1.0.1, Ubuntu 16.04
[Include Telegraf version, operating system name, and other relevant details]

### Steps to reproduce:
Add in a bad fcgi config, launch telegraf

## Expected behavior:
Not crashing

### Actual behavior:
Crashing

### Additional info:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x5f0530]

goroutine 63 [running]:
panic(0x12ecf20, 0xc8200100c0)
        /usr/lib/go-1.6/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc82060e090, 0xc8206080c0, 0xc8205b5c18, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/fcgi_client.go:40 +0x750
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200bfac0, 0xc82060e090, 0xc82021ebe8, 0x9, 0x7fd4ebce2840, 0xc82021e540, 0x0, 0x0)
        /build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:163 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200bfac0, 0xc82021ebd1, 0x20, 0x7fd4ebce2840, 0xc82021e540, 0x0, 0x0)
        /build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:150 +0x75e
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc82048bf30, 0xc82048bf40, 0xc8200bfac0, 0x7fd4ebce2840, 0xc82021e540, 0xc82021ebd1, 0x20)
        /build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
        /build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -61,6 +61,7 @@ continue sending logs to /var/log/telegraf/telegraf.log.
 - [#1731](https://github.com/influxdata/telegraf/issues/1731): Fix Graphite template ordering, use most specific.
 - [#1836](https://github.com/influxdata/telegraf/pull/1836): Fix snmp table field initialization for non-automatic table.
 - [#1724](https://github.com/influxdata/telegraf/issues/1724): cgroups path being parsed as metric.
+- [#1886](https://github.com/influxdata/telegraf/issues/1886): Fix phpfpm fcgi client panic when URL does not exist.
 
 ## v1.0.1 [2016-09-26]
 

--- a/plugins/inputs/phpfpm/phpfpm.go
+++ b/plugins/inputs/phpfpm/phpfpm.go
@@ -122,6 +122,9 @@ func (g *phpfpm) gatherServer(addr string, acc telegraf.Accumulator) error {
 		fcgiIp := socketAddr[0]
 		fcgiPort, _ := strconv.Atoi(socketAddr[1])
 		fcgi, err = newFcgiClient(fcgiIp, fcgiPort)
+		if err != nil {
+			return err
+		}
 		if len(u.Path) > 1 {
 			statusPath = strings.Trim(u.Path, "/")
 		} else {

